srcalc/src/main/java/gov/va/med/srcalc/db/ResultSearchParameters.java

```diff
@@ -0,0 +1,257 @@
+package gov.va.med.srcalc.db;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Restrictions;
+import org.joda.time.LocalDate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import gov.va.med.srcalc.domain.calculation.SignedResult;
+import gov.va.med.srcalc.util.SearchResults;
+
+/**
+ * <p>Encapsulates search parameters for {@link SignedResult}s.</p>
+ * 
+ * <p>Note: in most classes, we define the interface very tightly (using Optional objects
+ * instead of nulls, etc.), but here we are pretty flexible in what we accept to allow
+ * Spring to bind forms to these objects.</p>
+ */
+public final class ResultSearchParameters
```

Comment:
Shouldn't this class be labeled as a DAO or be using a smaller DAO class to execute the Criteria search in doSearch()? It seems to me that this class should hold the parameters and then another class has the database access method in it.